test/openai/test_completion.py

```diff
@@ -18,6 +19,50 @@
 from flaml.autogen.math_utils import eval_math_responses, solve_problem
 
 
+def yes_or_no_filter(context, response, **_):
+    return context["yes_or_no_choice"] is False or any(
+        text in ["Yes.", "No."] for text in oai.Completion.extract_text(response)
+    )
+
+
+def valid_json_filter(response, **_):
+    for text in oai.Completion.extract_text(response):
+        try:
+            json.loads(text)
+            return True
+        except ValueError:
+            pass
+    return False
+
+
+def test_filter():
+    try:
+        import openai
+    except ImportError as exc:
+        print(exc)
+        return
+    response = oai.Completion.create(
+        context={"yes_or_no_choice": True},
```

Comment:
Should also revise the docstr for `context`? 
E.g., Change
"It needs to contain keys that are used by the prompt template."
to 
"It needs to contain keys that are used by the prompt template and filter_func."